### DISCUSSION TITLE
Button Actions

### DISCUSSION THREAD

**Discussion by wlatic** [Budibase/budibase#1616] [Help]
Was running through a lot of the setup i needed for an internal application until I started using APIs.

Although I can send the API correctly within the admin setup I'm coming stuck when I have a Form, text field, button. The button will run the API based on the input in the text field.

I've tried all combinations of tables, data providers etc. and I just cant get the output to show anywhere!

**Comment by joebudi** [comment]
Thank you @wlatic . I am looking into this now.

**Comment by shogunpurple** [comment]
Hey @wlatic - so just to clarify:

- You have some external data sources set up with the REST connector
- You have a form, with a button and you use the "Execute Query" action on the button, so that you can submit the form and execute the query
- You want to take the response from this API call, and display it in the UI

If the above is correct, you cannot show the response from the query in the UI at this time. This is however a great suggestion - it would be good to be able to perform an action with the response from a query.

What would the best version of this look like for you? Do you just want to render a list of results from the response, or do something else with it?

**Reply by wlatic** [reply]
This makes me feel better. I thought I just couldn't get it.

I believe the majority of items coming from rest api are going to be arrays, so the table display would still work, possibly even linking the button to a Data element, then a table, to keep things consistent?

I'm sure there will be far bigger work flows on the back of buttons bringing data in.

**Reply by shogunpurple** [reply]
@wlatic nope, that one is on us! Great suggestion - I will convert this into a github issue, and we will raise it in our next product meetings. Thank you for raising!

**Reply by wlatic** [reply]
Is there any hacky way round this right now?

Custom code myself?

**Reply by ryzam** [reply]
Hi guys, I'm also having the same issue as mention by @wlatic . For example in my case when I click "Test Connection" button it will invoke rest API that return some data, unfortunately I can't inspect the data to perform disable/enable "Save" button. My suggestion to have another action type that can handle other element in the form.

![image](https://user-images.githubusercontent.com/597759/140610833-5e3de5d5-9335-4ca0-9030-eff726ca537c.png)

**Reply by wlatic** [reply]
@shogunpurple any update on this one? Even a work around would be great.

**Comment by wlatic** [comment]
Just been told on the reddit group that this has been implemented, is it true?

**Reply by mjashanks** [reply]
Possibly, depending on your exact needs.

When displaying data in the UI, you always need to use a data provider. We have added a "Refresh Data Provider" button action and "Conditional UI":

So, you could....
1. Create your input
2. Conditionally hide your data provider, until the input has a value
3. Show the data provider when input is entered - passing value into the data provider.

**Reply by rajendra233** [reply]
**Can't show the data in budibase from external drools API** .
I have created simple rule in drool project and I have pasted URL of the project in budibase RestAPI. I have used POST method and entered the data in the body in raw json and we are able to sucessfully hit the API. now is there any way to present data in budibase forms?
